### DISCUSSION TITLE
"Rename Tab" feature (a little explanation)

### DISCUSSION THREAD

**Discussion by EdoFro** [freeplane/freeplane#1490] [General]
# "Rename Tab" feature

The next preview version of Freeplane includes the ability to rename mind map view labels.

## Renaming single view tab

To rename a label just right-click on the label. A pop up menu will open where you have to click on "Rename tab".

https://github.com/freeplane/freeplane/assets/35700575/dba369ff-4a2f-4547-9952-be2aa2059092

## Reseting to original name

To return to the original name, simply leave the field empty or click on the "Cancel" button.

https://github.com/freeplane/freeplane/assets/35700575/040b2767-57d7-44b8-8307-3fc2ddfb4839

## Renaming "grouped Views" tab

Freeplane allows views to be grouped under a single label.

### Grouping views

To do this you must select a view and drag the others into that selected one.

**Difference between "changing layout" and "creating views groups":**

- normal case: "changing layout"
   - if you select a tab and drag'n drop it, then you can change the order of the tabs or the layout of the windows on the screen, but no group tab is created.
- "creating views groups", ( 3 or more views open)
   - if, instead, you drag'n drop another tab (not the selected one) and drop it into the selected view, then a new group is created that includes the selected view and the view you drop into it.

#### sample video:

1. I select view "CustomMenu" (click on its tab)
1. I drag'n drop tab "Mochila" into view "CustomMenu"
1. Now they are both part of a new group view
You can change its layout by moving the tabs inside the view

https://github.com/freeplane/freeplane/assets/35700575/8a65b0b5-23a0-4669-9727-98efebb436c2

### Renaming tab

You can also rename the labels of the grouped views.

After the new name, the number of views included is added in parenthesis

https://github.com/freeplane/freeplane/assets/35700575/bb0a669b-ae5d-4a56-824a-cac7c3309bac

## Known issues:

1. "Moving views inside a grouped view resets its tab text"
   - By changing the layout of the views within a group, its label can revert to its original name.
   - The external library used to manage the views creates new views each time a layout change is made and then deletes those views that are left over. In this process the original window can be replaced and its custom name disappears with it.
1. "When restarting Freeplane only single view tabs reloads their custom tabs names"
   - This is the way the external library works. 
1. "When restarting Freeplane only one of the views of a mindmap is opened, althought when closing Freeplane I had many opened."
   - This is the way the external library works.
1. "... and that one opens with the custom name of other that was not opened."
   - This is the way the external library works.

Hope it helps,

edo

**Comment by sifran-github** [comment]
Really nice.

Quick note:
> To return to the original name, simply leave the field empty or click on the "Cancel" button.

This is not the behavior I expect when doing such action.  When clicking cancel i expect it to .. cancel the operation and not change anything. and when leaving a field empty i also expect something similar, at least not to change anything.
I think that a better way is to have another option in the right click menu -> reset name. or even "reset to <default_name>".

**Comment by dpolivaev** [comment]
In the recent preview 1.11.8_05 Freeplane reopens all views and keeps their tab names.
It does not restore view specific filters and selected nodes.

Please test it and let me know if it works and if it helps.

https://sourceforge.net/projects/freeplane/files/freeplane%20preview/

**Reply by euu2021** [reply]
Sorry, I don't understand. I'm trying freeplane-1.11.8-pre05 here, and I see nothing different.

For example, If I have opened:

- Map1 - map view a - custom name A
- Map1 - map view b - custom name B
- Map2 - no custom name

If I restart FP, which of those 3 above should I be seeing? I'm seeing:
- Map 1 - no custom name

**Reply by dpolivaev** [reply]
@euu2021  Can you please upload some screenshots showing what you tried and what you observed?

**Reply by dpolivaev** [reply]
For example, if you never saved your map, the views can not be restored. It works only if your map has a related file.

**Reply by euu2021** [reply]
- Two maps opened, with two map views on each. All map views with custom names.
- I restart FP and get only the second map, with no custom name:

![7JDaNrGbBn](https://github.com/freeplane/freeplane/assets/77707706/06c16da2-7aa9-4674-a0ab-d790d3eec067)

**Reply by euu2021** [reply]
wait... I changed some preferences and it seems to be working now. I will test.

**Reply by euu2021** [reply]
It works, and it's perfect! Thanks!

The add-on [Freeplane_Map-Saving-Options](https://github.com/EdoFro/Freeplane_Map-Saving-Options), by Edo, would be a good companion for this, but it's giving null. I will report in the repository.
